overlord/snapstate/snapstate.go

```diff
@@ -1617,3 +1619,51 @@ func ConfigDefaults(st *state.State, snapName string) (map[string]interface{}, e
 
 	return defaults, nil
 }
+
+func refreshMisc(aStore StoreService) error {
+	e1 := refreshNames(aStore)
+	e2 := refreshSections(aStore)
+	if e1 != nil {
+		return e1
+	}
+	return e2
+}
+
+func refreshNames(aStore StoreService) error {
+	snapCmds, err := aStore.SnapCommands()
```

Comment:
Should we perhaps stream this instead of buffering it in memory. It may have essentially any size.